server/src/main/java/org/elasticsearch/monitor/metrics/NodeMetrics.java

```diff
@@ -212,6 +212,97 @@ private void registerAsyncMetrics(MeterRegistry registry) {
+                "es.node.stats.indices.indexing.docs.current",
+                "Current number of indexing documents",
+                "documents",
+                () -> new LongWithAttributes(stats.getOrRefresh().getIndices().getIndexing().getTotal().getIndexCurrent())
+            )
+        );
+
+        metrics.add(
+            registry.registerLongAsyncCounter(
+                "es.node.stats.indices.indexing.failed.total",
+                "Total number of failed indexing operations",
+                "operations",
+                () -> new LongWithAttributes(stats.getOrRefresh().getIndices().getIndexing().getTotal().getIndexFailedCount())
+            )
+        );
+
+        metrics.add(
+            registry.registerLongAsyncCounter(
+                "es.node.stats.indices.deletion.docs.total",
+                "Total number of deleted documents",
+                "documents",
+                () -> new LongWithAttributes(stats.getOrRefresh().getIndices().getIndexing().getTotal().getDeleteCount())
+            )
+        );
+
+        metrics.add(
+            registry.registerLongGauge(
+                "es.node.stats.indices.deletion.docs.current",
+                "Current number of deleting documents",
+                "documents",
+                () -> new LongWithAttributes(stats.getOrRefresh().getIndices().getIndexing().getTotal().getDeleteCurrent())
+            )
+        );
+
+        metrics.add(
+            registry.registerLongAsyncCounter(
+                "es.node.stats.indices.indexing.time",
+                "Total indices indexing time",
+                "milliseconds",
+                () -> new LongWithAttributes(stats.getOrRefresh().getIndices().getIndexing().getTotal().getIndexTime().millis())
+            )
+        );
+
+        metrics.add(
+            registry.registerLongAsyncCounter(
+                "es.node.stats.indices.deletion.time",
+                "Total indices deletion time",
+                "milliseconds",
+                () -> new LongWithAttributes(stats.getOrRefresh().getIndices().getIndexing().getTotal().getDeleteTime().millis())
+            )
+        );
+
+        metrics.add(
+            registry.registerLongAsyncCounter(
+                "es.node.stats.indices.throttle.time",
+                "Total indices throttle time",
+                "milliseconds",
+                () -> new LongWithAttributes(stats.getOrRefresh().getIndices().getIndexing().getTotal().getThrottleTime().millis())
+            )
+        );
+
+        metrics.add(
+            registry.registerLongAsyncCounter(
+                "es.node.stats.indices.noop.total",
+                "Total number of noop shard operations",
+                "operations",
+                () -> new LongWithAttributes(stats.getOrRefresh().getIndices().getIndexing().getTotal().getNoopUpdateCount())
+            )
+        );
+
+        metrics.add(
+            registry.registerLongAsyncCounter(
```

Comment:
Can we have all the indexing pressure stats exposed?

Reply:
Added. Yes, skipping indexing replication for now.